src/storage-preview/azext_storage_preview/_client_factory.py

```diff
@@ -129,3 +129,47 @@ def cf_blob_container_mgmt(cli_ctx, _):
 
 def cf_blob_data_gen_update(cli_ctx, kwargs):
     return blob_data_service_factory(cli_ctx, kwargs.copy())
+
+
+def get_account_url(cli_ctx, account_name, service):
+    from knack.util import CLIError
+    if account_name is None:
+        raise CLIError("Please provide storage account name or connection string.")
+    storage_endpoint = cli_ctx.cloud.suffixes.storage_endpoint
+    return "https://{}.{}.{}".format(account_name, service, storage_endpoint)
+
+
+def cf_share_service(cli_ctx, kwargs):
+    t_share_service = get_sdk(cli_ctx, CUSTOM_DATA_STORAGE_FILESHARE, '_share_service_client#ShareServiceClient')
+    connection_string = kwargs.pop('connection_string', None)
+    account_key = kwargs.pop('account_key', None)
+    token_credential = kwargs.pop('token_credential', None)
+    sas_token = kwargs.pop('sas_token', None)
+    account_name = kwargs.pop('account_name', None)
+    if connection_string:
+        return t_share_service.from_connection_string(conn_str=connection_string)
+
+    account_url = get_account_url(cli_ctx, account_name=account_name, service='file')
+    credential = account_key or sas_token or token_credential
+
+    if account_url and credential:
+        return t_share_service(account_url=account_url, credential=credential)
+    return None
+
+
+def cf_share_client(cli_ctx, kwargs):
+    return cf_share_service(cli_ctx, kwargs).get_share_client(share=kwargs.pop('share_name'),
+                                                              snapshot=kwargs.pop('snapshot', None))
+
+
+def cf_share_directory_client(cli_ctx, kwargs):
+    return cf_share_client(cli_ctx, kwargs).get_directory_client(directory_path=kwargs.pop('directory_path'))
+
+
+def cf_share_file_client(cli_ctx, kwargs):
+    return cf_share_client(cli_ctx, kwargs).get_file_client(file_path=kwargs.pop('file_path'))
+
+
+def cf_share_subdirectory_client(cli_ctx, kwargs):
```

Comment:
Is there any difference between subdirectory client and directory client?

Reply:
Yes, `share_directory_client` and `share_subdirectory_client` are only based on different `directory_path` and the logic of calling `ShareDirectoryClient` is the same.
I will delete the duplicate one~